tests/test_client.py

```diff
@@ -1196,3 +1202,97 @@ def test_debug_option(
+            issues_sampler=self.sampler_function_mock, sample_rate=self.sample_rate
+        )
+
+    def raise_exception(self):
+        # type: () -> None
+        raise self.exception_to_raise()
+
+
+@mock.patch("sentry_sdk.client.random.random", return_value=0.618)
+@pytest.mark.parametrize(
+    "test_config",
+    (
+        # Baseline test with issues_sampler only, both floats and bools
+        IssuesSamplerTestConfig(sampler_function=lambda _: 1.0, expected_events=1),
+        IssuesSamplerTestConfig(sampler_function=lambda _: 0.7, expected_events=1),
+        IssuesSamplerTestConfig(sampler_function=lambda _: 0.6, expected_events=0),
+        IssuesSamplerTestConfig(sampler_function=lambda _: 0.0, expected_events=0),
+        IssuesSamplerTestConfig(sampler_function=lambda _: True, expected_events=1),
+        IssuesSamplerTestConfig(sampler_function=lambda _: False, expected_events=0),
+        # Baseline test with sample_rate only
+        IssuesSamplerTestConfig(sample_rate=1.0, expected_events=1),
+        IssuesSamplerTestConfig(sample_rate=0.7, expected_events=1),
+        IssuesSamplerTestConfig(sample_rate=0.6, expected_events=0),
+        IssuesSamplerTestConfig(sample_rate=0.0, expected_events=0),
+        # issues_sampler takes precedence over sample_rate
+        IssuesSamplerTestConfig(
+            sampler_function=lambda _: 1.0, sample_rate=0.0, expected_events=1
+        ),
+        IssuesSamplerTestConfig(
+            sampler_function=lambda _: 0.0, sample_rate=1.0, expected_events=0
+        ),
+        # Different sample rates based on exception
+        IssuesSamplerTestConfig(
+            sampler_function=lambda event: {
+                "ZeroDivisionError": 1.0,
+                "AttributeError": 0.0,
+            }[event["exception"]["values"][0]["type"]],
+            exception_to_raise=ZeroDivisionError,
+            expected_events=1,
+        ),
+        IssuesSamplerTestConfig(
+            sampler_function=lambda event: {
+                "ZeroDivisionError": 1.0,
+                "AttributeError": 0.0,
+            }[event["exception"]["values"][0]["type"]],
+            exception_to_raise=AttributeError,
+            expected_events=0,
+        ),
+    ),
+)
+def test_issues_sampler(_, sentry_init, capture_events, test_config):
+    test_config.init_sdk(sentry_init)
+
+    events = capture_events()
+
+    try:
+        test_config.raise_exception()
+    except Exception:
+        capture_exception()
+
+    assert len(events) == test_config.expected_events
+
+    try:
+        assert test_config.sampler_function_mock.call_count == 1
+
+        # Ensure one argument (the event) was passed to the sampler function
+        assert len(test_config.sampler_function_mock.call_args[0]) == 1
+    except AttributeError:
+        # sampler_function_mock should be None in this case,
+        # but let's double-check to ensure the test is working correctly
+        assert test_config.sampler_function_mock is None
```

Comment:
I find this easier to read:
```suggestion
    if test_config.sampler_function_mock is not None:
        assert test_config.sampler_function_mock.call_count == 1

        # Ensure one argument (the event) was passed to the sampler function
        assert len(test_config.sampler_function_mock.call_args[0]) == 1
```